### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by osvaldoFernandez92**
Summary

Sign up: 
Trello card solved: https://trello.com/c/J9VDZuY9
Angular user controller implemented: It does some validations, such as verify password length and that "confirm password" matches "password" field.

API request made with RestAngular. JSON response used in console.log

### CODE REVIEW

**Comment by osvaldoFernandez92** [comment]
@sbalay

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+node_modules
+bower_components
+npm-debug.log
+build
+package.json
```
Comment (sbalay) [inline .gitignore]: Remove package.json from gitignore

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+.ruby-version
+npm-shrinkwrap.json
+development.js
```
Comment (sbalay) [inline .gitignore]: We usually don't add development.js keys to gitignore.

File: package.json [hunk 3]
```diff
@@ -44,6 +44,7 @@
     "gulp-sourcemaps": "^1.5.2",
     "gulp-uglify": "^1.2.0",
     "jshint-stylish": "^1.0.1",
+    "lodash": "^3.10.1",
```
Comment (sbalay) [inline package.json]: Why lodash here?

File: src/app/app.routes.js [hunk 4]
```diff
@@ -1,42 +1,50 @@
 app.config([
-  '$stateProvider', '$urlRouterProvider', '$locationProvider',
-  function($stateProvider, $urlRouterProvider, $locationProvider) {
+  '$stateProvider', '$urlRouterProvider', '$locationProvider', 'RestangularProvider','configuration',
+  function($stateProvider, $urlRouterProvider, $locationProvider, RestangularProvider, configuration) {
+
+  // Restangular configuration
+    RestangularProvider.setBaseUrl(configuration.baseParseURL);
+    RestangularProvider.setDefaultHeaders({
```
Comment (sbalay) [inline src/app/app.routes.js]: Set this default headers to the `$http` service.

Comment (sbalay) [inline src/app/app.routes.js]: https://docs.angularjs.org/api/ng/service/$http#setting-http-headers

File: src/app/app.routes.js [hunk 5]
```diff
@@ -1,42 +1,50 @@
...
         }
       }
     })
-    .state('state1.index', {
-      url: '/state1',
+    .state('home.index', {
+      url: '/home',
       views: {
         innerComponent: {
           templateUrl: '../app/components/component1/component1.html'
```
Comment (sbalay) [inline src/app/app.routes.js]: Rename the template

File: src/app/app.routes.js [hunk 6]
```diff
@@ -1,42 +1,50 @@
...
         }
       }
     })
-    .state('state2.index', {
-      url: '/state2',
+    .state('sign_up.index', {
+      url: '/sign_up',
       views: {
         innerComponent: {
           templateUrl: '../app/components/component2/component2.html'
```
Comment (sbalay) [inline src/app/app.routes.js]: rename the template

File: src/app/components/component1/component1Controller.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
 angular.module('app').controller('Component1Controller', [
```
Comment (sbalay) [inline src/app/components/component1/component1Controller.js]: rename the controller

File: src/app/components/component2/component2.jade [hunk 8]
```diff
@@ -1,5 +1,25 @@
 div(ng-controller="Component2Controller as ctrl2")
```
Comment (sbalay) [inline src/app/components/component2/component2.jade]: rename the controller

File: src/app/components/component2/component2.jade [hunk 9]
```diff
@@ -1,5 +1,25 @@
...
   h1.frontend-bootstrap-component-2
     {{ctrl2.component2Phrase}}
-  a(ui-sref='state1.index')
-    | Go to state1
+  form(name='userForm' ng-controller='UserController as userCtrl' ng-submit='userForm.$valid && userCtrl.addUser()' novalidate)
+    | Name (*)
+    input(name='name', ng-model='userCtrl.user.name', required)
+    br
```
Comment (sbalay) [inline src/app/components/component2/component2.jade]: Remove these `br` tags. Separate these inputs using css.

File: src/app/components/component2/component2.jade [hunk 10]
```diff
@@ -1,5 +1,25 @@
...
+    | Last Name (*)
+    input(name='lastname', ng-model='userCtrl.user.lastname', required)
+    br
+    | Username (*)
+    input(name='usrname', ng-model='userCtrl.user.usrname', required)
```
Comment (sbalay) [inline src/app/components/component2/component2.jade]: typo usErname

File: src/app/components/component2/component2Controller.js [hunk 11]
```diff
@@ -1,4 +1,49 @@
 angular.module('app').controller('Component2Controller', [
```
Comment (sbalay) [inline src/app/components/component2/component2Controller.js]: rename

File: src/app/components/component2/component2Controller.js [hunk 12]
```diff
@@ -1,4 +1,49 @@
...
 function() {
-  this.component2Phrase = 'This is component 2';
+  this.component2Phrase = 'Register';
 }]);
+
+
+angular.module('app').controller('UserController', ['Restangular',
```
Comment (sbalay) [inline src/app/components/component2/component2Controller.js]: use different files for each controller

File: src/app/components/component2/component2Controller.js [hunk 13]
```diff
@@ -1,4 +1,49 @@
...
+var compareTo = function() {
+    return {
+        require: "ngModel",
+        scope: {
+            otherModelValue: "=compareTo"
+        },
+        link: function(scope, element, attributes, ngModel) {
+             
+            ngModel.$validators.compareTo = function(modelValue) {
+                return modelValue == scope.otherModelValue;
```
Comment (sbalay) [inline src/app/components/component2/component2Controller.js]: always compare with `===`

File: src/app/components/component2/component2Controller.js [hunk 14]
```diff
@@ -1,4 +1,49 @@
...
+	this.user={};
+	this.addUser=function(){
+		//NEW USER
+		var newUser = {
+  			"username": this.user.usrname,
+			"password": this.user.pw1,
+			"email": this.user.email,
+			"firstName": this.user.name,
+			"lastName": this.user.lastname
+		}
```
Comment (sbalay) [inline src/app/components/component2/component2Controller.js]: missing semicolon and bad identation. check the linter output in the console.

**Comment by sbalay** [comment]
@osvaldoFernandez92 Work on these comments and then I'll review this again.

**Comment by osvaldoFernandez92** [comment]
@sbalay  fixes applied

**Comment by sbalay** [comment]
:+1:

**Comment by sbalay** [comment]
@osvaldoFernandez92 close or merge this pull request.

**Pull request closed without merging**